src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -796,25 +1031,27 @@ def forward(
             position_ids = torch.arange(past_length, input_shape[-1] + past_length, dtype=torch.long, device=device)
             position_ids = position_ids.unsqueeze(0)
 
-        # GPT2Attention mask.
+        # Attention mask.
         if attention_mask is not None:
-            if batch_size <= 0:
-                raise ValueError("batch_size has to be defined and > 0")
+            assert batch_size > 0, "batch_size has to be defined and > 0"
```

Comment:
Library convention is no asserts in modeling files 

```suggestion
```

Reply:
There are a bunch of `assert`s in the modeling file of `GPT2` which I suppose are legacy